src/shiftentercommand.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @module enter/shiftentercommand
+ */
+
+import Command from '@ckeditor/ckeditor5-core/src/command';
+
+/**
+ * ShiftEnter command. It is used by the {@link module:enter/shiftenter~ShiftEnter ShiftEnter feature} to handle
+ * the <kbd>Shift</kbd>+<kbd>Enter</kbd> keystroke.
+ *
+ * @extends module:core/command~Command
+ */
+export default class ShiftEnterCommand extends Command {
+	/**
+	 * @inheritDoc
+	 */
+	execute() {
+		const model = this.editor.model;
+		const doc = model.document;
+
+		model.change( writer => {
+			softBreakAction( model, writer, doc.selection );
+			this.fire( 'afterExecute', { writer } );
+		} );
+	}
+
+	refresh() {
+		const model = this.editor.model;
+		const doc = model.document;
+
+		this.isEnabled = isEnabled( model.schema, doc.selection );
+	}
+}
+
+// Checks whether the shiftEnter command should be enabled in the specified selection.
+//
+// @param {module:engine/model/schema~Schema} schema
+// @param {module:engine/model/selection~Selection|module:engine/model/documentselection~DocumentSelection} selection
+function isEnabled( schema, selection ) {
+	if ( selection.rangeCount > 1 ) {
+		return false;
+	}
+
+	const anchorPos = selection.anchor;
+
+	// Check whether the break element can be inserted in the current selection anchor.
+	if ( !anchorPos || !schema.checkChild( anchorPos, 'softBreak' ) ) {
+		return false;
+	}
+
+	const range = selection.getFirstRange();
+	const startElement = range.start.parent;
+	const endElement = range.end.parent;
+
+	// If the selection contains at least two elements and one of them is the limit element, the soft enter shouldn't be enabled.
+	if ( ( schema.isLimit( startElement ) || schema.isLimit( endElement ) ) && startElement !== endElement ) {
```

Comment:
This logic is incomplete. Selection can be deeply nested inside limit elements. E.g.:

```html
<limit>
    <paragraph>X[Y</paragrap>
</limit>
<paragraph>Z]A</paragraph>
```

Besides, I'm not sure whether this command must be disabled in this case. It can be because cross-limit selections are (most likely) to be banned at all. But in any case, it's good to explain such decisions.